### PR TITLE
Document CA1071: Override base.Equals when implementing IEquatable

### PR DESCRIPTION
**Pull request by manfred-brands**
## Summary

Added documentation for new rule CA1071: Override base.Equals if implementing IEquatable.

Belongs to roslyn analyzer PR: https://github.com/dotnet/roslyn-analyzers/pull/3992

### CODE REVIEW

File: docs/fundamentals/code-analysis/quality-rules/ca1071.md [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+the base class <xref:System.IEquatable%601.Equals%2A> method.
+
+## Rule description
+
+A type implementing <xref:System.IEquatable%601> interface indicates that it
+supports comparing two instances of the type for equality. You should also
+override the base class implementation of <xref:System.IEquatable%601.Equals%2A>
+method so that its behavior is consistent with that of the
+<xref:System.IEquatable%601.Equals%2A?displayProperty=fullName> implementation.
+See [here](/dotnet/api/system.iequatable-1#notes-to-implementers) for more
```
Comment (Youssef1313) [inline docs/fundamentals/code-analysis/quality-rules/ca1071.md:33]:
```suggestion
For more information, see [Notes to implementers](/dotnet/api/system.iequatable-1#notes-to-implementers).
```

**Review by gewarren — CHANGES_REQUESTED** [review]
Thanks for adding this article. Please also add it to the [TOC.yml file](https://github.com/dotnet/docs/blob/master/docs/fundamentals/toc.yml) and the [index page](https://github.com/dotnet/docs/blob/master/docs/fundamentals/code-analysis/quality-rules/design-warnings.md).

File: docs/fundamentals/code-analysis/quality-rules/ca1071.md [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+        : base(value)
+    {
+        Name = name;
+    }
+
+    public string Name { get; }
+
+    public bool Equals(C other) => base.Equals(other) && Name == other.Name;
+
+    public sealed override bool Equals(B other) => Equals(other as C);
```
Comment (gewarren) [inline docs/fundamentals/code-analysis/quality-rules/ca1071.md:108]:
```suggestion
    // This line fixes the violation.
    public sealed override bool Equals(B other) => Equals(other as C);
```

File: docs/fundamentals/code-analysis/quality-rules/ca1071.md [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+        Name = name;
+    }
+
+    public string Name { get; }
+
+    public bool Equals(C other) => base.Equals(other) && Name == other.Name;
+}
+```
+
+```csharp
```
Comment (gewarren) [inline docs/fundamentals/code-analysis/quality-rules/ca1071.md:79]:
```suggestion
The following code snippet shows the same two classes with the violation addressed.

```csharp
```

**Comment by gewarren** [comment]
[Preview link (internal)](https://review.docs.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/ca1071?branch=pr-en-us-21838).

**Comment by gewarren** [comment]
Oops, looks like it also needs to be added to https://github.com/manfred-brands/docs/blob/master/docs/fundamentals/code-analysis/quality-rules/index.md, sorry.

**Review by gewarren — APPROVED** [review]

**Pull request merged**